tests/erlang_tests/test_integer_to_list.erl

```diff
@@ -56,3 +62,27 @@ compare_list([H_A | T_A], [H_B | T_B]) when H_A == H_B ->
     compare_list(T_A, T_B);
 compare_list(_A, _B) ->
     0.
+
+test_bases() ->
+    "0" = integer_to_list(0),
+    "0" = erlang:integer_to_list(0, 10),
+    "-1" = integer_to_list(-1),
+    "-1010" = erlang:integer_to_list(-10, 2),
+    "1010" = erlang:integer_to_list(10, 2),
+    "A" = erlang:integer_to_list(10, 16),
+    "123456789ABCDEF0" = erlang:integer_to_list(16#123456789ABCDEF0, 16),
+    "A" = erlang:integer_to_list(10, 36),
+    assert_badarg(fun() -> erlang:integer_to_list(10, 1) end),
+    assert_badarg(fun() -> erlang:integer_to_list(10, 0) end),
+    assert_badarg(fun() -> erlang:integer_to_list(10, -1) end),
+    assert_badarg(fun() -> erlang:integer_to_list(10, 37) end),
```

Comment:
Does `erlang:integer_to_list` gets compiled to an actual call to `erlang:integer_to_list`, or does it gets replaced with a constant?

Reply:
`erlang:integer_to_list/2` is compiled to a call to the nif, however integer_to_list/1 of a constant was optimized. I fixed that.

Also, I did write `erlang:integer_to_list/2` in the tests because of bad habits based on historical reasons. At some point, one had to explicitely mention the module name when calling `integer_to_list/2` as opposed to `integer_to_list/1`. This is no longer true with OTP21 and I amended the test.